baseplate/ratelimit.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import division
+
+import time
+
+
+class RateLimitExceededException(Exception):
+    """This exception gets raised whenever a rate limit is exceeded.
+    """
+    pass
+
+
+class RateLimiter:
+    """A class for rate limiting actions.
+
+    :param `RateLimitCache` cache: The backend
+        to use for storing rate limit counters.
+    :param int allowance: The maximum allowance allowed per key.
+    :param int interval: The interval (in seconds) to reset allowances.
+    :param str key_prefix: A prefix to add to keys during rate limiting.
+        This is useful if you will have two different rate limiters that will
+        receive the same keys.
+
+    """
+
+    def __init__(self, cache, allowance=None, interval=None, key_prefix=''):
+        assert allowance >= 1, 'minimum allowance is 1'
+        assert interval >= 1, 'minimum interval is 1'
+
+        self.cache = cache
+        self.key_prefix = key_prefix
+        self.allowance = allowance
+        self.interval = interval
+
+    def consume(self, key, amount=1):
+        """Consume the given `amount` from the allowance for the given `key`.
+
+        This will rate
+        :py:class:`baseplate.ratelimit.RateLimitExceededException` if the
+        allowance for `key` is exhausted.
+
+        :param str key: The name of the rate limit bucket to consume from.
+        :param int amount: The amount to consume from the rate limit bucket.
+
+        """
+        key = self.key_prefix + key
+        if not self.cache.consume(key, amount, self.allowance, self.interval):
+            raise RateLimitExceededException('Rate limit exceeded.')
+
+
+class RateLimitCache:
```

Comment:
Since baseplate also supports python2, you should include the `(object)` here (you don't need it in python3 but it changes some behavior when you don't have it in python2).